src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+        Math_Vec3f_Copy(&this->unk2BC, &vec3fcopy);
+        this->actor.flags |= 0x1; // targetable ON
+        this->actionFunc = EnNiw_Thrown;
+    }
+    func_80891320(this, globalCtx, 2);
+}
+
+// action function: recently thrown, and also hopping on the floor
+void EnNiw_Thrown(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->unk2EC == 0) {
+        if ((this->actor.bgCheckFlags & 1)) { // about to hit the floor
+            this->unk2EC = 1;
+            this->unkTimer252 = 80; // hop timer
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 4.0f;
+        } else {
+            return; // wait until back on floor
+        }
+    } else {
+        if ((this->actor.bgCheckFlags & 1)) { // about to hit the floor
+            this->sfxTimer1 = 0;
+            this->actor.velocity.y = 4.0f; // vertical hop
+            this->unk29E = 1;
+        }
+        if (this->unkTimer252 == 0) {
+            this->unkTimer254 = 100;
+            this->unkTimer250 = 0;
+            this->unk2EC = 0;
+            EnNiw_SetupRunning(this);
+            return;
+        }
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        // picked up again before could run off
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+        this->sfxTimer1 = 30;
+        this->unk2EC = 0;
+        this->unkTimer250 = 30;
+        this->actor.flags &= ~0x1; // targetable OFF
+        this->unknownState28E = 4;
+        this->actionFunc = EnNiw_Held;
+        this->actor.speedXZ = 0.0f;
+    } else {
+        if (this->unkTimer252 >= 6) {
+            func_800B8BB0(&this->actor, globalCtx);
+        }
+        func_80891320(this, globalCtx, 2);
+    }
+}
+
+// EnNiw_Swimming
+// action func: swimming and flying away after swimming
+void EnNiw_Swimming(EnNiw* this, GlobalContext* globalCtx) {
+    Vec3f ripplePos;
+
+    // even if hitting water, keep calling for reinforcements
+    // this should just be in update
+    if (this->niwStormActive) {
+        EnNiw_SpawnAttackNiw(this, globalCtx); // spawn attack niw
+    }
+
+    this->actor.speedXZ = 2.0f;
+    if ((this->actor.bgCheckFlags & 0x20)) {
+        // still touching water
+        this->actor.gravity = 0.0f;
+        if (this->actor.yDistToWater > 15.0f) {
+            this->actor.world.pos.y = this->actor.world.pos.y + 2.0f;
+        }
+        if (this->unkTimer250 == 0) {
+            this->unkTimer250 = 30;
+            Math_Vec3f_Copy(&ripplePos, &this->actor.world.pos);
+            ripplePos.y += this->actor.yDistToWater;
+
+            EffectSsGRipple_Spawn(globalCtx, &ripplePos, 100, 500, 30);
+        }
+        if ((this->actor.bgCheckFlags & 8) != 0) {
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 1.0f;
+        }
+    } else {
+        this->actor.gravity = -2.0f;
+        if ((this->actor.bgCheckFlags & 8)) {
+            // has hit a wall
```

Comment:
```suggestion
        if (this->actor.bgCheckFlags & 8) {
```
Same as my comment above for `bgCheckFlags`